### ISSUE TITLE
Container layout method should be called ndlayout

### ISSUE THREAD

**Issue by jlstevens** [holoviz/holoviews#1606]
The ``.layout`` method of containers such as ``HoloMap`` should be renamed to ``.ndlayout`` as this method returns an ``NdLayout`` and *not* a ``Layout``.

**Edit**: Same logic applies to the ``overlay`` method: it should be called ``ndoverlay``.

**Comment by jbednar** [comment]
This naming has also confused me in the past, but I'm not sure .ndlayout and .ndoverlay would be better, as they are much harder to remember and to read in source code.  Is it better to be precise but obscure, or to be a bit misleading but clear on the main point (that the result will be laid out or overlaid)?

**Comment by philippjfr** [comment]
I'm -1 on this, it's been around forever and I favor the simpler name. Unless you ever want to add ``.overlay`` and ``.layout`` methods that cast to actual Overlays and Layouts (which I don't consider very useful) my vote is to keep things as they are.

**Comment by jlstevens** [comment]
> Unless you ever want to add .overlay and .layout methods that cast to actual Overlays and Layouts.

I don't see why not.

``def layout(self):
       return Layout(self.values())
```

Then people can choose what they want and use a method that is named appropriately for the thing it returns.

**Comment by philippjfr** [comment]
At this point that would be highly confusing and I honestly don't think it should be encouraged because NdOverlay and NdLayout are the correct datastructures for homogeneous data.

**Comment by jlstevens** [comment]
I don't feel super strongly about changing it but everytime I use ``.layout`` I have to do a double-take to remind myself that I get an ``NdLayout`` and not a ``Layout``.

I suppose you can pretend the method is a verb indicating an imperative action 'layout' as opposed to a noun, in which case you can 'layout' your items in different ways (as a ``Layout`` or ``NdLayout``). The same logic applies to 'overlay' though I do think this naming will continue to cause occasional confusion...

In short, unless @jbednar weighs in on this and supports this suggestion, I don't mind leaving it as it is because I no longer see it as incorrect when considered as a verb instead of as a noun...

**Comment by jbednar** [comment]
I'd prefer to leave it as it is, all things considered.  We can encourage the verb interpretation in our docs, and mention that it's an NdLayout (and why).

**Comment by jlstevens** [comment]
> We can encourage the verb interpretation in our docs, and mention that it's an NdLayout (and why).

Good plan! I'll think I'll leave this issue open as a remind to do just that...

**Comment by jlstevens** [comment]
This ship sailed a long time ago and nowadays I think the `layout` method is a nicer name, even if it brushes over this technicality.
